pkg/controllers/vrep/vdbverify_reconciler.go

```diff
@@ -0,0 +1,91 @@
+package vrep
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	v1beta1 "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	vrepstatus "github.com/vertica/vertica-kubernetes/pkg/vrepstatus"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const stateIncompatibleDB = "Incompatible"
+
+type VdbVerifyReconciler struct {
+	VRec *VerticaReplicatorReconciler
+	Vrep *v1beta1.VerticaReplicator
+	Log  logr.Logger
+}
+
+func MakeVdbVerifyReconciler(r *VerticaReplicatorReconciler, vrep *v1beta1.VerticaReplicator,
+	log logr.Logger) controllers.ReconcileActor {
+	return &VdbVerifyReconciler{
+		VRec: r,
+		Vrep: vrep,
+		Log:  log.WithName("VdbVerifyReconciler"),
+	}
+}
+
+// Reconcile will verify the verticaDBs in the vrep CR source and target exist and
+// both vertica versions support the replication feature
+func (r *VdbVerifyReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	// no-op if the check has already been done once
+	isSet := r.Vrep.IsStatusConditionTrue(v1beta1.ReplicationReady)
+	if isSet {
+		return ctrl.Result{}, nil
+	}
+
+	vdbSource := &vapi.VerticaDB{}
+	vdbTarget := &vapi.VerticaDB{}
+	nmSource := names.GenNamespacedName(r.Vrep, r.Vrep.Spec.Source.VerticaDB)
+	nmTarget := names.GenNamespacedName(r.Vrep, r.Vrep.Spec.Target.VerticaDB)
+	if res, err := vk8s.FetchVDB(ctx, r.VRec, r.Vrep, nmSource, vdbSource); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+	if res, err := vk8s.FetchVDB(ctx, r.VRec, r.Vrep, nmTarget, vdbTarget); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+
+	// check version for both vdbs, the minimim version should be 24.3.0
+	vinfSource, err := vdbSource.MakeVersionInfoCheck()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	vinfTarget, err := vdbTarget.MakeVersionInfoCheck()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if !vinfSource.IsEqualOrNewer(vapi.ReplicationSupportedMinVersion) {
```

Comment:
I assume the 2 vdbs are not required to have the same version?

Reply:
I don't know whether we need to enforce the source and target to have the same version. The JIRA ticket only mentions that we need to check whether both meet the minimum version requirement. I think we need more info on the version requirement for source and target `vdb`s here. @chinhtranvan do you have more info?

Reply:
I don't think it's necessary to require the same version for both the source and target databases. The main idea is to upgrade all nodes in a database (or sandbox) first, and then replicate to another database (or sandbox). In the sandbox, it will be a separate cluster that no longer sees modifications from the main cluster. Therefore, the version may differ from the main cluster.